### DISCUSSION TITLE
Configuration in Log Analysis Language

### DISCUSSION THREAD

**Discussion by Terni** [apache/skywalking#9728] [General]
Hello skywalking team. 

I need help with Log Analysis Language. This is the first time I work with it.
I progressed by documentation
https://skywalking.apache.org/docs/main/v9.2.0/en/concepts-and-designs/lal/

My question:
**How to catch some log with fatal or error level and send email on our support team?**
 
I try set configuration in LAL but i do not know is corectrly or not. Next point how to sent to our support team via email? 

For example 

```
filter {
    text {
        abortOnFailure true // this is optional because it's default behaviour
        // this is just a demo pattern
        regexp "(?<timestamp>\\d{8}) (?<thread>\\w+) (?<level>\\w+) (?<traceId>\\w+) (?<msg>.+)"
    }
    extractor {
        service parsed.serviceName
        tag level: parsed.level, (parsed.statusCode): parsed.statusMsg
        // metrics {
        //     name "log_count"
        //     timestamp parsed.timestamp
        //     labels level: parsed.level, service: parsed.service, instance: parsed.instance
        //     value 1
        // }
    }
    sink {
        if (parsed.level == "DEBUG") {
            dropper {}
        } else if (parsed.level == "FATAL") {
            sampler {
                if (parsed.service == "Dashborad-Service") {
                    possibility(80) { 
                    }
                }
            }
        } else if (parsed.level == "ERROR") {
            sampler {
                if (parsed.service == "Dashborad-Service") {
                    possibility(30) { 
                    }
                }
            }
        }
    }
}
```

**Comment by wu-sheng** [comment]
LAL is to help you structure logs and extra metrics. Sending an email is not its responsibility, so, no, there isn't that feature. The key point is, sending a mail base on a single log is not affordable in a prod env. Your email server and OPS team could easily be drown out

**Comment by kezhenxu94** [comment]
Maybe extract metrics for the log level or log pattern and configure alarm for that metrics.

**Comment by Terni** [comment]
Yes I mean metrics as wrote @kezhenxu94.

- firstly must extract information from log level with using LAL and evaluate them
- after i want send information email ( i do not list of logs) meaby reports only

Is it possible through an alarm or something?
https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-alarm/#default-alarm-rules

**Reply by wu-sheng** [reply]
We don't have email notification directly, we could use HTTP or gRPC hook to build one and forward the notification.

**Reply by Terni** [reply]
Thanks, I'll try it. Maybe it's later, I'll ask for help.

**Comment by Terni** [comment]
Please help. 
I tried x-times different configuration in LAL but not work any configuration file.

My last simple version is:
```
rules:
  - name: test-rules
    layer: GENERAL
    dsl: |
      filter {
          extractor {
              service parsed.serviceName
              tag level: parsed.level
              // we add a tag called `level` and its value is parsed.level, captured from the regexp above
              traceId parsed.traceId
              // we also extract the trace id from the parsed result, which will be used to associate the log with the trace
              metrics {
                  name "log_count"
                  timestamp parsed.timestamp
                  labels level: parsed.level, service: parsed.service, instance: parsed.instance
                  value 1
              }
          }
          // ...
      }
```
This file i saved into /skywalking/config/lal/test-rules.yaml

and I changed section in application.yml file
```
log-analyzer:
  selector: ${SW_LOG_ANALYZER:default}
  default:
    lalFiles: ${SW_LOG_LAL_FILES:test-rules}
    malFiles: ${SW_LOG_MAL_FILES:""}
```

Where will I see any result in the UI? Is it a serviceMesh section?
![image](https://user-images.githubusercontent.com/1767477/194897275-ee258685-d16a-406c-b325-5fecdcf7c86a.png)

I read the whole document about LAL but I still don't know how to configure it properly.

**Reply by wu-sheng** [reply]
What do you want to see? You neither call `sink` for load, nor set up MAL script for metrics. 
I can't follow your expectation. 

There are many logs analyses in the release, also, we have provided a showcase for a quick start.

Why would not you at least make sure logs are collected first? Then add new LAL script step by step?
When you are not familiar with something, don't change everything, and ask why it is not working. No one knows, especially, we are just a community. We can't afford the time to debug and set up for you step by step.

**Reply by Terni** [reply]
Ok, I'll try to solve it myself.
